File: src/ros2_bridge/mappers/control_msgs/joint_tolerance.rs
//! Typed mapper for `control_msgs/msg/JointTolerance`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn joint_tolerance_to_bus(msg: ros_env::control_msgs::msg::JointTolerance) -> crate::control_msgs::msg::v1::JointTolerance {
    crate::control_msgs::msg::v1::JointTolerance {
        name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.name),
        position: msg.position,
        velocity: msg.velocity,
        acceleration: msg.acceleration,
    }
}

pub(crate) fn joint_tolerance_to_ros(bus: crate::control_msgs::msg::v1::JointTolerance) -> ros_env::control_msgs::msg::JointTolerance {
    ros_env::control_msgs::msg::JointTolerance {
        name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.name),
        position: bus.position,
        velocity: bus.velocity,
        acceleration: bus.acceleration,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct ControlMsgsJointToleranceMapper;

impl TypedTopicMapper for ControlMsgsJointToleranceMapper {
    type Ros = ros_env::control_msgs::msg::JointTolerance;
    type Bus = crate::control_msgs::msg::v1::JointTolerance;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(joint_tolerance_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(joint_tolerance_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/control_msgs/joint_tolerance.py
"""Generated mapper for `control_msgs/msg/JointTolerance`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.control_msgs.msg.v1 import JointTolerance as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def joint_tolerance_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.name = str(msg.name)
    bus.position = msg.position
    bus.velocity = msg.velocity
    bus.acceleration = msg.acceleration
    return bus


def joint_tolerance_to_ros(bus):
    from control_msgs.msg import JointTolerance as RosMsg

    out = RosMsg()
    out.name = str(bus.name)
    out.position = bus.position
    out.velocity = bus.velocity
    out.acceleration = bus.acceleration
    return out


class ControlMsgsJointToleranceMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from control_msgs.msg import JointTolerance as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return joint_tolerance_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return joint_tolerance_to_ros(bus)
